crates/ruff_linter/src/rules/flake8_simplify/rules/enumerate_for.rs

```diff
@@ -0,0 +1,178 @@
+use ruff_python_ast::{self as ast, Constant, ExceptHandler, Expr, Int, MatchCase, Operator, Stmt};
+use ruff_text_size::{Ranged, TextRange};
+
+use crate::rules::flake8_simplify::rules::ast_bool_op::is_same_expr;
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::traversal;
+
+use crate::checkers::ast::Checker;
+
+/// ## What it does
+/// Checks for loops which has explicit loop-index variables that can be simplified by using `enumerate()`.
+///
+/// ## Why this is bad?
+/// Using `enumerate()` is more readable and concise. It could lead to more efficient
+/// and less error-prone code.
+///
+/// ## Example
+/// ```python
+/// sum = 0
+/// idx = 0
+/// for item in items:
+///    sum += func(item, idx)
+///   idx += 1
+/// ```
+///
+/// Use instead:
+/// ```python
+/// sum = 0
+/// for idx, item in enumerate(items):
+///    sum += func(item, idx)
+/// ```
+///
+/// ## References
+/// - [Python documentation: enumerate()](https://docs.python.org/3/library/functions.html#enumerate)
+#[violation]
+pub struct EnumerateForLoop {
+    index_var: String,
+}
+
+impl Violation for EnumerateForLoop {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        let EnumerateForLoop { index_var } = self;
+        format!("Use enumereate() for index variable `{index_var}` in for loop")
```

Comment:
Small typo in the output recommendation, but an important one since it's the name of the replacement function.

The snapshots need this change too.

```suggestion
        format!("Use enumerate() for index variable `{index_var}` in for loop")
```